### DISCUSSION TITLE
How can i use local default swiper?

### DISCUSSION THREAD

**Discussion by Neliver** [nolimits4web/swiper#5666] [Q&A] [answered]
I have found default css(swiper.min.css), but i can't find default js, only bundle. Where can i find default js-file and singly plug modules, which i need?

**Comment by Arhell** [comment, marked as answer]
https://swiperjs.com/get-started

**Reply by Neliver** [reply]
Ты, как я понял, понимаешь по-русски. Либо ты не прочитал вопрос, либо я не смог нормально спросить на английском. На странице которую ты скинул предлагаются три способа подключение: npm, cdn и дают ссылку на скачивание для локального подключения. Я хочу подключить локально, но только базовую версию слайдера, а необходимые модули добавить отдельно. Но я не смог найти файл с дефолтным js, только с дефолтным css. Причём если подключать через npm, то там ты как раз так и делаешь: подключаешь дефолт и к нему добавляешь нужные модули. Т.е. где-то этот файл должен быть.

**Reply by Arhell** [reply]
Привет, сразу напишу я не тестировал, но предпологаю что при npm установке можно найти нужный тебе файл.
https://swiperjs.com/get-started#install-from-npm

Или вот еще вариант 
https://github.com/nolimits4web/swiper/blob/master/src/swiper.js#L2

**Reply by nolimits4web** [reply]
> Ты, как я понял, понимаешь по-русски. Либо ты не прочитал вопрос, либо я не смог нормально спросить на английском. На странице которую ты скинул предлагаются три способа подключение: npm, cdn и дают ссылку на скачивание для локального подключения. Я хочу подключить локально, но только базовую версию слайдера, а необходимые модули добавить отдельно. Но я не смог найти файл с дефолтным js, только с дефолтным css. Причём если подключать через npm, то там ты как раз так и делаешь: подключаешь дефолт и к нему добавляешь нужные модули. Т.е. где-то этот файл должен быть. 

Только так https://swiperjs.com/swiper-api#using-build-script

**Reply by Neliver** [reply]
> Только так https://swiperjs.com/swiper-api#using-build-script

это как я понимаю как раз примерно то же самое, что и предложил Arhell? Просто немного меняется порядок.  Ну, я понял что придётся с npm разбираться в любом случае, ни разу с ним не работал ещё. А имеет вообще смысл подключать локально только часть функционала, или лишний код не будет влиять на скорость загрузки так как не будет использоваться?

**Reply by Neliver** [reply]
> Привет, сразу напишу я не тестировал, но предпологаю что при npm установке можно найти нужный тебе файл. https://swiperjs.com/get-started#install-from-npm
> 
> Или вот еще вариант https://github.com/nolimits4web/swiper/blob/master/src/swiper.js#L2

Не совсем понимаю как работает вторая ссылка, но исходя из неё core.js это как раз то, что нужно. Но я пытался подключать core.js и даже если вначале убрать кучу импортов, то всё равно ошибку выдавало, уже не вспомню какую.
